src/main/java/com/openlattice/chronicle/services/surveys/SurveysManagerImpl.java

```diff
@@ -0,0 +1,467 @@
+import static com.openlattice.chronicle.constants.EdmConstants.VALUES_FQN;
+import static com.openlattice.chronicle.util.ChronicleServerUtil.checkNotNullUUIDs;
+import static com.openlattice.chronicle.util.ChronicleServerUtil.getFirstUUIDOrNull;
+import static com.openlattice.edm.EdmConstants.ID_FQN;
+
+/**
+ * @author alfoncenzioka &lt;alfonce@openlattice.com&gt;
+ */
+public class SurveysManagerImpl implements SurveysManager {
+    protected static final Logger logger = LoggerFactory.getLogger( SurveysManagerImpl.class );
+
+    private final ApiCacheManager       apiCacheManager;
+    private final EnrollmentManager     enrollmentManager;
+    private final CommonTasksManager    commonTasksManager;
+    private final ScheduledTasksManager scheduledTasksManager;
+
+    public SurveysManagerImpl(
+            ApiCacheManager apiCacheManager,
+            EnrollmentManager enrollmentManager,
+            CommonTasksManager commonTasksManager,
+            ScheduledTasksManager scheduledTasksManager ) {
+        this.apiCacheManager = apiCacheManager;
+        this.enrollmentManager = enrollmentManager;
+        this.commonTasksManager = commonTasksManager;
+        this.scheduledTasksManager = scheduledTasksManager;
+    }
+
+    @Override
+    public ChronicleQuestionnaire getQuestionnaire(
+            UUID organizationId, UUID studyId, UUID questionnaireEKID ) {
+        try {
+            logger.info( "Retrieving questionnaire: orgId = {} studyId = {}, questionnaire EKID = {}",
+                    organizationId,
+                    studyId,
+                    questionnaireEKID );
+
+            UUID studyEKID = Preconditions
+                    .checkNotNull( commonTasksManager.getStudyEntityKeyId( organizationId, studyId ),
+                            "invalid study: " + studyId );
+
+            // get apis
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            SearchApi searchApi = apiClient.getSearchApi();
+
+            // entity set ids
+            UUID questionnaireESID = commonTasksManager
+                    .getEntitySetId( organizationId, CHRONICLE_SURVEYS, SURVEY, QUESTIONNAIRE_ES );
+            UUID studiesESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, STUDIES, STUDY_ES );
+            UUID partOfESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, PART_OF, PART_OF_ES );
+            UUID questionESID = commonTasksManager
+                    .getEntitySetId( organizationId, CHRONICLE_SURVEYS, QUESTION, QUESTIONS_ES );
+
+            checkNotNullUUIDs( ImmutableSet.of( questionnaireESID, studiesESID, partOfESID, questionESID ) );
+
+            // Get questionnaires that neighboring study
+            Map<UUID, List<NeighborEntityDetails>> neighbors = searchApi.executeFilteredEntityNeighborSearch(
+                    studiesESID,
+                    new EntityNeighborsFilter(
+                            Set.of( studyEKID ),
```

Comment:
these should all be ImmutableSet.of( blah )

Reply:
https://github.com/openlattice/chronicle-server/pull/143#discussion_r516948672